lightning/src/ln/funding.rs

```diff
@@ -20,69 +20,92 @@ use crate::sign::{P2TR_KEY_PATH_WITNESS_WEIGHT, P2WPKH_WITNESS_WEIGHT};
 
 /// The components of a splice's funding transaction that are contributed by one party.
 #[derive(Debug, Clone)]
-pub enum SpliceContribution {
-	/// When funds are added to a channel.
-	SpliceIn {
-		/// The amount to contribute to the splice.
-		value: Amount,
-
-		/// The inputs included in the splice's funding transaction to meet the contributed amount
-		/// plus fees. Any excess amount will be sent to a change output.
-		inputs: Vec<FundingTxInput>,
-
-		/// An optional change output script. This will be used if needed or, when not set,
-		/// generated using [`SignerProvider::get_destination_script`].
-		///
-		/// [`SignerProvider::get_destination_script`]: crate::sign::SignerProvider::get_destination_script
-		change_script: Option<ScriptBuf>,
-	},
-	/// When funds are removed from a channel.
-	SpliceOut {
-		/// The outputs to include in the splice's funding transaction. The total value of all
-		/// outputs plus fees will be the amount that is removed.
-		outputs: Vec<TxOut>,
-	},
+pub struct SpliceContribution {
+	/// The amount to contribute to the splice.
+	value: SignedAmount,
+
+	/// The inputs included in the splice's funding transaction to meet the contributed amount
+	/// plus fees. Any excess amount will be sent to a change output.
+	inputs: Vec<FundingTxInput>,
```

Comment:
Should we just leave these fields public as they were (implicitly cause it was an enum)? Could then presumably drop the `splice_in_and_out` constructor?

Reply:
Hmm... I wouldn't say it's a net contribution for the splice-in portion. It's just that a `FundingTxInput` doesn't specify how much of it we should spend -- and probably shouldn't since more than one can be given. Whereas for the splice-out portion there isn't a parallel, so there's no need for the user to duplicately specify the amount and risk miscalculating it.